cmake/Modules/FindPythonExtra.cmake

```diff
@@ -17,211 +17,75 @@
-    message(STATUS "Using PythonExtra_LIBRARIES: ${PythonExtra_LIBRARIES}")
-  else()
-    find_package(PythonLibs 3.5 REQUIRED)
-    if(WIN32 AND CMAKE_BUILD_TYPE STREQUAL "Debug")
-      get_filename_component(_python_executable_dir "${PYTHON_EXECUTABLE}" DIRECTORY)
-      get_filename_component(_python_executable_name "${PYTHON_EXECUTABLE}" NAME_WE)
-      get_filename_component(_python_executable_ext "${PYTHON_EXECUTABLE}" EXT)
-      set(_python_executable_debug "${_python_executable_dir}/${_python_executable_name}_d${_python_executable_ext}")
-      if(EXISTS "${_python_executable_debug}")
-        set(PYTHON_EXECUTABLE_DEBUG "${_python_executable_debug}")
-      else()
-        message(FATAL_ERROR "${_python_executable_debug} doesn't exist")
-      endif()
-    endif()
-    message(STATUS "Using PYTHON_EXECUTABLE: ${PYTHON_EXECUTABLE}")
-    message(STATUS "Using PYTHON_INCLUDE_DIRS: ${PYTHON_INCLUDE_DIRS}")
-    message(STATUS "Using PYTHON_LIBRARIES: ${PYTHON_LIBRARIES}")
-    set(PythonExtra_INCLUDE_DIRS "${PYTHON_INCLUDE_DIRS}")
-    set(PythonExtra_LIBRARIES "${PYTHON_LIBRARIES}")
-  endif()
-
-  if(NOT DEFINED PYTHON_SOABI)
-    set(_python_code
-      "from sysconfig import get_config_var"
-      "print(get_config_var('SOABI'))"
-    )
-    execute_process(
-      COMMAND
-      "${PYTHON_EXECUTABLE}"
-      "-c"
-      "${_python_code}"
-      OUTPUT_VARIABLE _output
-      RESULT_VARIABLE _result
-      OUTPUT_STRIP_TRAILING_WHITESPACE
-    )
-    if(NOT _result EQUAL 0)
-      message(FATAL_ERROR
-        "execute_process(${PYTHON_EXECUTABLE} -c '${_python_code}') returned "
-        "error code ${_result}")
-    endif()
-
-    set(PYTHON_SOABI
-      "${_output}"
-      CACHE INTERNAL
-      "The SOABI suffix for Python native extensions. See PEP-3149: https://www.python.org/dev/peps/pep-3149/.")
-  endif()
-
-  if(PYTHON_SOABI STREQUAL "" OR PYTHON_SOABI STREQUAL "None")
-    set(PythonExtra_EXTENSION_SUFFIX
-      ""
-      CACHE INTERNAL
-      "The full suffix for Python native extensions. See PEP-3149: https://www.python.org/dev/peps/pep-3149/."
-    )
-  else()
-    set(PythonExtra_EXTENSION_SUFFIX
-      ".${PYTHON_SOABI}"
-      CACHE INTERNAL
-      "The full suffix for Python native extensions. See PEP-3149: https://www.python.org/dev/peps/pep-3149/."
-    )
-  endif()
+if(PythonExtra_FOUND)
+  return()
+endif()
 
-  if(WIN32)
-    if(CMAKE_BUILD_TYPE STREQUAL "Debug")
-      set(PythonExtra_EXTENSION_EXTENSION "_d.pyd")
-    else()
-      set(PythonExtra_EXTENSION_EXTENSION ".pyd")
-    endif()
-  else()
-    # Also use .so for OSX, not dylib
-    set(PythonExtra_EXTENSION_EXTENSION ".so")
+find_package(Python3 REQUIRED COMPONENTS Interpreter)
+
+get_target_property(PYTHON_EXECUTABLE Python3::Interpreter LOCATION)
+
+add_executable(PythonExtra::Interpreter IMPORTED)
+set_property(TARGET PythonExtra::Interpreter
+  PROPERTY IMPORTED_LOCATION "${PythonExtra_EXECUTABLE}")
```

Comment:
Where is `PythonExtra_EXECUTABLE` set?

Reply:
Nowhere, this was a bug.  Thanks for pointing it out, fixed ing e74ed3b.